renku/cli/dataset.py

```diff
@@ -618,45 +644,63 @@ def import_(ctx, client, uri, name, extract):
             click.secho('OK', fg='green')
 
 
-def download_file(position, extract, data_folder, file, chunk_size=16384):
+def download_file(extract, data_folder, file, chunk_size=16384):
     """Download a file with progress tracking."""
+    global current_process_position
+
     local_filename = Path(file.filename).name
     download_to = Path(data_folder) / Path(local_filename)
 
-    def extract_dataset(data_folder_, filename, file_):
+    def extract_dataset(data_folder_, filename):
         """Extract downloaded dataset."""
         import patoolib
-        filepath = data_folder_ / filename
+        filepath = Path(data_folder_) / Path(filename)
         patoolib.extract_archive(filepath, outdir=data_folder_)
         filepath.unlink()
 
     def stream_to_file(request):
         """Stream bytes to file."""
         with open(download_to, 'wb') as f_:
+            scaling_factor = 1e-6
+            unit = 'MB'
+
+            if file.filesize < 1e5:
```

Comment:
Can we add a comment on what this math is referring to? Just as a reference for the future.